scripts/update_files.py

```diff
@@ -0,0 +1,51 @@
+from pathlib import Path
+import re
+import os
+import sys
+
+def check_tex_file_exists(repo_name: str, base_path: Path):
+    """Check if a .tex file with the same name as the repository exists."""
+    tex_file_path = base_path / f'{repo_name}.tex'
+    if not tex_file_path.exists():
+        sys.exit(f"Error: No .tex file found with the name {repo_name}.tex")
+
+def read_file(file_path: Path) -> str:
+    """Reads the entire content of a file and returns it as a string."""
+    return file_path.read_text()
+
+def write_file(file_path: Path, content: str):
+    """Writes the given content to a file, replacing its previous content."""
+    file_path.write_text(content)
+
+def update_readme_content(content: str, repo_name: str) -> str:
+    """Updates the README.md content."""
+    content = re.sub(r'^.*', f'# {repo_name}', content, count=1)
+    actions_status_line = f'[![Actions Status](https://github.com/{os.getenv("GITHUB_REPOSITORY")}/workflows/CI/badge.svg)](https://github.com/{os.getenv("GITHUB_REPOSITORY")})'
```

Comment:
Let's pass the repo name as argument to this function rather than accessing the env var twice.